<!DOCTYPE html>
<html>
<head>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.1.1/jquery.min.js"></script>
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
    <link rel="stylesheet" href="welcome_project.css">
    <meta charset='UTF-8'>
    <title>Assembly gaps</title>
</head>
<body>
<div class='content'>
    <div class ="tabs">
        <ul class ="nav nav-tabs" id="tab-links">
            <li class="active" value="tab0"><a data-toggle="tab" href="#menu0">Assembly gaps</a></li>
            <li  value="tab1"><a data-toggle="tab" href="#menu1">List of other projects</a></li>
            <li  value="tab2"><a data-toggle="tab" href="#menu2">About</a></li>
        </ul>
    </div>
    <div class="tab-content">
        <div id="menu0" class="tab-pane fade  in active">
            <div class="container">
                <div class="rowSpecial">
                    <div class="colAll">
                        <h2 class="projectTitle">Assembly gaps</h2>
                        <div class = 'project'>
                            <div class = 'projectDesc'>
                                <p class="paragraph">
                                    <div><b>Background</b></div>
                                    The difficulties associated with sequencing and assembling some regions of the DNA sequence result in gaps in the reference genomes that are typically represented as stretches of Ns. Although the presence of assembly gaps causes slight reduction in the mapping rate in many experimental settings, that does not invalidate the typical statistical testing comparing read count distributions across experimental conditions. However, we hypothesize that not handling assembly gaps in the null model may confound statistical testing of co-localization of genomic features.
                                </p>
                                <p class="paragraph">
                                    <div><b>Results</b></div>
                                    First, we performed a series of <b>explorative analyses</b> to understand whether and how the public genomic tracks intersect the assembly gaps track (hg19).
                                    The findings rightly confirm that the genomic regions in public genomic tracks intersect very little with assembly gaps and the intersection was observed only at the beginning and end regions of the assembly gaps rather than covering the whole gap sizes.
                                    Further, we simulated a set of query and reference genomic tracks in a way that nullified any dependence between them to test our hypothesis that not avoiding assembly gaps in the null model would result in spurious inflation of statistical significance.
                                We then <b>contrasted the distributions of test statistics and p-values of Monte Carlo simulation-based permutation tests that either avoided or not avoided assembly gaps in the null model when testing for significant co-localization between a pair of query and reference tracks</b>.
                                    We observed that the statistical tests that did not account for the assembly gaps in the null model resulted in a distribution of p-values that is shifted to the left (leading to inflated significance) and a distribution of the test statistic that is shifted to the right.
                                </p>
                                <p class="paragraph">
                                    <div><b>Conclusion</b></div>
                                    Our results confirm that not accounting for assembly gaps in statistical testing of co-localization analysis leads to false positives and over-optimistic findings.
                                </p>
                                <p class="paragraph">
                                    To read more about <b>'Mind your gaps: Overlooking of assembly gaps confounds statistical testing in genome analysis'</b> go to: TO BE WRITTEN LATER.
                                </p>
                                <!-- <p class="paragraph">https://hyperbrowser.uio.no/assemblygaps/u/hb-superuser/p/assembly-gaps
                                To see all examples from above paper <a href="../u/hb-superuser/p/assembly-gaps">click here</a>..
                                </p> -->
                            </div>
                        </div>
                    </div>
                </div>
            </div>
            <div class="containerNext">
                <div class="rowSpecialWithGap">
                    <div class="colAll">
                        <div class = 'description'>
                        <h4 class = 'descriptionTitle'>Overlap of public genomic tracks with genome assembly gaps</h4>
                            <div class = 'descriptionDesc'>
                                <p class="paragraph">
                                    We performed a series of explorative analyses to understand the nature and extent of intersection of genomic tracks with genome assembly gaps.
                                </p>
                                <p class="paragraph">
                                    All the examples were executed based on 3 large collections of public genomics tracks:
                                    <ul>
                                        <li><a href="../u/hb-superuser/h/dataset-histmod" target="_blank">histone modifications collection</a>, which consist of 477 tracks</li>
                                        <li><a href="../u/hb-superuser/h/analysis-dnase" target="_blank">DNase hypersensitive sites collection</a>, which consist of 838 tracks</li>
                                        <li><a href="../u/hb-superuser/h/dataset-k562" target="_blank">transcription factor bindingsites in K562 cell line collection</a>, which consist of 568 tracks</li>
                                    </ul>
                                </p>
                                <p class="paragraph">
                                    The following examples are based on dataset: DNase hypersensitive sites collection and <a href="../u/hb-superuser/h/dataset-assembly-gaps" target="_blank"> assembly gaps track (element 1) and start, middle and end region of assembly gaps track (element 2, 3, 4, respectively) </a>.
                                    The rest of the analysis described in the paper 'Mind your gaps: Overlooking of assembly gaps confounds statistical testing in genome analysis' can be found <a href="../u/hb-superuser/p/assembly-gaps" target="_blank">here.</a>
                                </p>
                            </div>
                        </div>
                        <div class = 'example'>
                            <h4 class = 'exampleTitle'>
                                Number of elements per track in GSuite
                            </h4>
                            <div class = 'exampleDesc'>
                                The following tool: "Compute a basic measure for each track in a GSuite" is used to count number of elements per every track in DNase hypersensitive sites collection.
                                <a href="../tool_runner/data_source_redirect?tool_id=hb_g_suite_single_value_analysis_per_track_tool">Click here to use the tool</a>
                                or
                                <a href = "../u/hb-superuser/h/analysis-dnase">here to view an example (element 14, 19)</a>
                            </div>
                            <div class = 'exampleDesc'>
                                The following tool: "Create a plot for selected columns in a tabular file" is used to visualize the results of the above calculations.
                                <a href="../tool_runner/data_source_redirect?tool_id=hb_plot_metadata_values_of_tabular_file_tool">Click here to use the tool</a>
                                or
                                <a href = "../u/hb-superuser/h/analysis-dnase">here to view an example (element 20)</a>
                            </div>
                        </div>
                        <div class = 'example'>
                            <h4 class = 'exampleTitle'>
                                Average segment length of track and bp proportion overlap value (with selected track) per every track in a collection of tracks (GSuite)
                            </h4>
                            <div class = 'exampleDesc'>
                                The following tool: "Count average segment length of track and bp proportion overlap value (with selected track) per every track in a collection of tracks (GSuite)" is used to count average segment length track and bp proportion overlap value per every track in DNase hypersensitive sites collection and assembly gaps track.
                                <a href="../tool_runner/data_source_redirect?tool_id=hb_count_avg_segment_length_and_bp_overlap_tool">Click here to use the tool</a>
                                or
                                <a href = "../u/hb-superuser/h/analysis-dnase">here to view an example (element 15, 16)</a>
                            </div>
                            <div class = 'exampleDesc'>
                                The following tool: "Create a plot for selected columns in a tabular file" is used to visualize the results of the above calculations.
                                <a href="../tool_runner/data_source_redirect?tool_id=hb_plot_metadata_values_of_tabular_file_tool">Click here to use the tool</a>
                                or
                                <a href = "../u/hb-superuser/h/analysis-dnase">here to view an example (element 17, 18)</a>
                            </div>
                        </div>
                        <div class = 'example'>
                            <h4 class = 'exampleTitle'>
                                Intersection between the genomic tracks (collection of tracks - GSuite) and selected track
                            </h4>
                            <div class = 'exampleDesc'>
                                The following tool: "Count coverage for every track within specified region" is used to count intersection between the DNase hypersensitive sites collection and assembly gaps track.
                                <a href="../tool_runner/data_source_redirect?tool_id=hb_generate_visualization_overlap_heatmap_tool">Click here to use the tool</a>
                                or
                                <a href = "../u/hb-superuser/h/analysis-dnase-vs-assemblygapstrack-bp-proportion">here to view an example (element 23, 24)</a>
                                The following tool: "Create descriptive results for tabular file" is used to present results of the above calculations.
                                <a href="../tool_runner/data_source_redirect?tool_id=hb_create_histogram_tool">Click here to use the tool</a>
                                or
                                <a href = "../u/hb-superuser/h/analysis-dnase-vs-assemblygapstrack-bp-proportion">here to view an example (element 31)</a>
                            </div>
                            <div class = 'exampleDesc'>
                                The same tool: "Count coverage for every track within specified region" is used to count intersection between the DNase hypersensitive sites collection and start region of assembly gaps track.
                                <a href = "../u/hb-superuser/h/analysis-dnase-vs-assemblygapstrack-bp-proportion">Click here to view an example (element 25, 26)</a>
                                The following tool: "Create descriptive results for tabular file" is used to present results of the above calculations.
                                <a href="../tool_runner/data_source_redirect?tool_id=hb_create_histogram_tool">Click here to use the tool</a>
                                or
                                <a href = "../u/hb-superuser/h/analysis-dnase-vs-assemblygapstrack-bp-proportion">here to view an example (element 32)</a>
                            </div>
                            <div class = 'exampleDesc'>
                                The same tool: "Count coverage for every track within specified region" is used to count intersection between the DNase hypersensitive sites collection and middle region of assembly gaps track.
                                <a href = "../u/hb-superuser/h/analysis-dnase-vs-assemblygapstrack-bp-proportion">Click here to view an example (element 27, 28)</a>
                                The following tool: "Create descriptive results for tabular file" is used to present results of the above calculations.
                                <a href="../tool_runner/data_source_redirect?tool_id=hb_create_histogram_tool">Click here to use the tool</a>
                                or
                                <a href = "../u/hb-superuser/h/analysis-dnase-vs-assemblygapstrack-bp-proportion">here to view an example (element 33)</a>
                            </div>
                            <div class = 'exampleDesc'>
                                The same tool: "Count coverage for every track within specified region" is used to count intersection between the DNase hypersensitive sites collection and end region of assembly gaps track.
                                <a href = "../u/hb-superuser/h/analysis-dnase-vs-assemblygapstrack-bp-proportion">Click here to view an example (element 29, 30)</a>
                                The following tool: "Create descriptive results for tabular file" is used to present results of the above calculations.
                                <a href="../tool_runner/data_source_redirect?tool_id=hb_create_histogram_tool">Click here to use the tool</a>
                                or
                                <a href = "../u/hb-superuser/h/analysis-dnase-vs-assemblygapstrack-bp-proportion">here to view an example (element 34)</a>
                            </div>
                            <div class = 'exampleDesc'>
                                Also, average segment length of track and bp overlap value (with selected track) per every track in a collection of tracks (GSuite) were counted in the same way. The results are presented <a href = "../u/hb-superuser/h/analysis-dnase-vs-assemblygapstrack-bp">here</a>.
                            </div>
                        </div>
                    </div>
                </div>
            </div>

            <div class="containerNext">
                <div class="rowSpecialWithGap">
                    <div class="colAll">
                        <div class = 'description'>
                        <h4 class = 'descriptionTitle'>The impact of not avoiding the assembly gaps on the findings of statistical testing in co-localization analysis</h4>
                            <div class = 'descriptionDesc'>
                                <p class="paragraph">
                                    We tested the significance of pairwise overlap of 100 pairs of genomic tracks by accounting and not accounting for assembly gaps.
                                </p>
                            </div>
                        </div>
                        <div class = 'example'>
                            <h4 class = 'exampleTitle'>
                                Preparing data for MC-based permutation tests either avoided or not avoided assembly gaps in the null model.
                            </h4>
                            <div class = 'exampleDesc'>
                                The following tool: "Sample tracks from a collection of tracks (GSuite)" creates a collection of tracks (GSuite) with specified number of tracks (randomly selected from DNase hypersensitive sites collection).
                                <a href="../tool_runner/data_source_redirect?tool_id=hb_randomize_gsuite_tool">Click here to use the tool</a>
                                or
                                <a href = "../u/hb-superuser/h/randomization---gsuite-dnase">here to view an example (element 30)</a>
                            </div>
                            <div class = 'exampleDesc'>
                                The following tool: "Randomize a collection of genomic tracks (GSuite)" is used to shuffles genomic tracks by excluding the randomization in certain regions (assembly gaps track).
                                 <a href="../tool_runner/data_source_redirect?tool_id=hb_count_null_model_for_randomized_g_suite_tool">Click here to use the tool</a>
                                or
                                <a href = "../u/hb-superuser/h/randomization---gsuite-dnase">here to view an example (element 30-36)</a>
                            </div>

                        </div>
                    </div>
                </div>

                     <div class="colAll">
                        <div class = 'descriptionDesc'>
                            <p class="paragraph">
                                The following tool: "Paired co-localization analysis of two collections of tracks" is used to count Forbes or Forbes with a p-val with the selected MCFDR sampling depth and with or without excluded regions track.
                                <a href="../tool_runner/data_source_redirect?tool_id=hb_categorical_g_suite_vs_g_suite_tool">Click here to use the tool</a>.
                            </p>
                        </div>
                     </div>


                <div class="">
                    <div class="colHalfLeft">
                        <div class = 'example'>
                            <h4 class = 'exampleTitle'>
                                Testing the significance of pairwise overlap without exclusion of assembly gaps track.
                            </h4>
                            <div class = 'exampleDesc'>
                                <a href = "../u/hb-superuser/h/hb_categorical_g_suite_vs_g_suite_tool">Click here to view an example (element 10)</a>.
                            </div>
                        </div>
                    </div>
                    <div class="colHalfRight" style="border-left:0px solid transparent;">
                        <div class = 'example'>
                            <h4 class = 'exampleTitle'>
                                Testing the significance of pairwise overlap with exclusion of assembly gaps track
                            </h4>
                            <div class = 'exampleDesc'>
                                <a href = "../u/hb-superuser/h/hb_categorical_g_suite_vs_g_suite_tool">Click here to view an example (element 11)</a>.
                            </div>
                        </div>
                    </div>
                </div>
            </div>

            <div class="footer">
                <div class = 'contactInfo'>
                    <div class = 'contactInfoDesc'>
                        <p class="paragraph">
                            If you want to read more about The Genomic HyperBrowser interface, <a href="https://sites.google.com/site/hyperbrowserhelp/" target="_blank"> click here </a>
                        </p>
                        <p class="paragraph">
                            If you have any questions, requests or comments regarding the assembly gaps project, please contact: <a href='mailto:dianadom@ifi.uio.no' target='_top'> Diana Domanska</a> or <a href='mailto:geirksa@ifi.uio.no' target='_top'>Geir Kjetil Sandve</a>.
                        </p>
                    </div>
                </div>
            </div>
        </div>
        <div id="menu1" class="tab-pane fade ">
            <div class="container">
                <div class="rowSpecial">
                    <div class="colAllProjectList">
                        <div id ='content'></div>
                            <script>
                            $("#content").load("/dianadom_dev/static/hyperbrowser/files/projects/projects_assemblygaps.txt");
                            </script>
                    </div>
                </div>
            </div>
        </div>
        <div id="menu2" class="tab-pane fade ">
            <div class="container">
                <div class="contact">
                    <b>Note:</b>
                        <p>
                        This is a specialized version of The Genomic Hyperbrowser,
                        focusing on new functionality for analyzing collections of genomic track.
                        More information on the general HyperBrowser functionality can be
                        found on the <a href="https://hyperbrowser.uio.no/hb/" target="_blank">main project site</a>.
                        </p>
                    <b>Contact information</b>
                        <p>
                        If you have any questions, requests or comments, please use our discussion forum (available from the <a minsizehint="-1" href="https://sites.google.com/site/hyperbrowserhelp/?tool_id=hb_help">Help</a> menu) or send us an email to <a href="#" onclick="this.href='mailto:'+ this.innerHTML.split('').reverse().join('')" class="codedirection">hyperbrowser-requests@usit.uio.no</a>&#x200E;. We are happy to respond and welcome collaborative projects that help further develop our system. If you encounter any bugs, please let us know via an email to
                        <a href="#" onclick="this.href='mailto:'+ this.innerHTML.split('').reverse().join('')" class="codedirection">hyperbrowser-requests@usit.uio.no</a>.
                       </p>

                        <p>If you register as a user, you will be able to subscribe to the <a href="#" onclick="this.href='mailto:'+ this.innerHTML.split('').reverse().join('')" class="codedirection">hyperbrowser-requests@usit.uio.no</a> mailing list, which serves for distribution of system-related announcements and administrative messages.
                         A backlog of announcements is available from the Help menu. The Help menu also contains citation information.
                        </p>
                </div>
            </div>

            <div class="containerNext">
                <div class="rowSpecial">
                    <div class ="infoAboutHB">
                    <p class="blocktext">
                    This project is being developed by the Norwegian bioinformatics community as an open-source project under the GPL  license v3,
                    supported by various national and local bodies.
                    </p>
                    <p class="blocktext">
                      For further information on Norwegian bioinformatics, please visit <a href="http://www.bioinfo.no">www.bioinfo.no</a>.
                    </p>
                    </div>
                </div>
            </div>


            <div class="footer">
                <div class="section">
                    <img src="hyperbrowser/images/logo/HB_support.png" class="center">
                </div>
            </div>
        </div>
    </div>
</div>
</body>
</html>