<!DOCTYPE html>
<head><meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.1.1/jquery.min.js"></script>
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
    <link rel="stylesheet" href="welcome_project.css">
    <meta charset='UTF-8'>
    <title>Colocalization analysis: bringing differential analysis to genome-scale omics data</title>
</head>
<body>
<div class='content'>
    <div class ="tabs">
        <ul class ="nav nav-tabs" id="tab-links">
            <li class="active" value="tab0"><a data-toggle="tab" href="#menu0">Project page</a></li>
            <li  value="tab1"><a data-toggle="tab" href="#menu1">User Guide</a></li>
            <li  value="tab2"><a data-toggle="tab" href="#menu2">List of other projects</a></li>
            <li  value="tab3"><a data-toggle="tab" href="#menu3">About</a></li>
        </ul>
    </div>
    <div class="tab-content">
        <div id="menu0" class="tab-pane fade  in active">
            <div class="container">
                <div class="rowSpecial">
                    <div class="colAll">
                        <h2 class="projectTitle">
                            Colocalization analysis: bringing differential analysis to genome-scale omics data
                        </h2>
                        <p class="paragraph">
                            A broad variety of chip- and high throughput sequencing-based methods produce sets of genomic regions as their main output.
                            Hence, the statistical analysis of genomic colocalization is a fundamental approach for integrative genomics.
                            Since individual genomic datasets may be affected by noise, experimental artifacts or particularities of individual donors or cell types,
                            it would be preferable to search for signals associated with sets of related experimental datasets. However, such analyses are challenging to
                            devise due to the many dependencies along the genome and across related genomic features, and no principled methodology is currently available for such investigations.


                        </p>
                        <p class="paragraph">
                        We here consider the question of whether two sets of genomic datasets differ significantly in their degree of colocalization with a reference genomic feature.
                            For instance, the two sets could be two active chromatin sets, and the reference feature GWAS SNPs, so the investigation of interest would be in which chromatin
                            set the GWAS SNPs are more active. The problem involves two main challenges; the often strong and varying degree of dependence between related (epi-)genomic features,
                            and the general hidden confounders influencing both the expected value and variance of colocalization between datasets. We define three benchmarks that together reflect
                            these main challenges. We propose several alternative analytical approaches and investigate their behavior on the defined benchmarks, as well as applying those methods on a
                            real biological case. We show that a non parametric approach based on a rank-order test statistic perform adequately on all three defined benchmarks.
                        </p>

                        <p class="paragraph">
                            The GitHub repository with the full functionality underlying the web-provided tools is available at <a href="https://github.com/hyperbrowser/genomic-hyperbrowser/tree/pr_rand_ts/group_test"  target="_blank">tools functionality</a>.
                        </p>

                        <p class="paragraph">
                              Data preparation and all results on simulated and real data are described below. For more details go to related sections <b>Method</b> and <b>Results</b> in the paper (to be updated).
                        </p>


                    </div>
                </div>
            </div>
            <div class="containerNext">
                <div class="rowSpecialWithGap">
                  <div class="colAll">

                      <h2 class="projectTitle">
                          Benchmarks:
                      </h2>
                        The figure below shows the randomization procedure adopted in each benchmark.

                        <div class = 'example'>
                            <div class = 'exampleDesc'>
                            <div class="main">
                                 <div class="onlyTop">
                                        <div class="imgzoom">
                                            <img src="../static/project/BMs.png" style="float:left;width: 100%; height: 20%; margin-right:10px;" />
                                        </div>
                                 </div>
                                    <div class="onlyRight" style="margin-left:10px;">

                                    </div>

                                </div>
                            </div>
                        </div>

                    </div>
                </div>
            </div>
            <div class="containerNext">
                <div class="rowSpecialWithGap">
                  <div class="colAll">

                      <h2 class="projectTitle">
                          Benchmark 1 - Existence of an empirical correlation structure across case tracks and control tracks

                      </h2>


                        <div class = 'example'>
                            <div class = 'exampleDesc'>
                            <div class="main">

                                    All the results of this benchmark are available
                                    at the following <a href="../u/hbsuperuser/h/21-gdt--benchmark-one"  target="_blank">link</a> you can find the 100 generated tracks (element 1).
                                    Elements 9, 11, 13, 25, 27 show the resulting p-values obtained when applying the Wilcoxon, MC1-3 and the t-test respectively.

                                </div>
                            </div>
                        </div>

                    </div>
                </div>
            </div>

            <div class="containerNext">
                <div class="rowSpecialWithGap">
                  <div class="colAll">

                      <h2 class="projectTitle">
                          Benchmark 2 - Existence of an increased level of colocalization for both case and control tracks against the reference
                      </h2>
                      <div class = 'exampleDesc'>
                          <div class="main">


                                    All the results of this benchmark are available
                                    at the following <a href="../u/hbsuperuser/h/22-gdt---benchmark-two"  target="_blank">link</a>.
                                    Element 8 in the history is the reference single track, while elements 18, 24, 43, 22 are the tracks simulated under benchmark 2, assuming no difference between
                                    the two sets of tracks and a signal factor of 1.01, 1.02 and 1.05 respectively. The p-values obtained by applying the different methods on these simulated datasets
                                      are shown by elements
                                    <p class="paragraph">
                                    The resulting p-values obtained by applying the different methods on these simulated tracks are available at the same link.

                                    </p>

                          </div>
                      </div>
                  </div>
                </div>
            </div>



            <div class="containerNext">
                <div class="rowSpecialWithGap">
                  <div class="colAll">

                      <h2 class="projectTitle">
                          Benchmark 3: Existence of increased level of colocalization and non-differentiating correlation structure
                      </h2>
                      <div class = 'example'>

                            <div class = 'exampleDesc'>
                      <div class="main">



                                All the results of this benchmark are available
                                    at the following <a href="../u/hbsuperuser/h/23-gdt---benchmark-three"  target="_blank">link</a> you can find the 100 tracks with randomized labels (element 6) and the single reference track (element 4).
                                Elements 9, 47, 39, 49, 51 show the resulting p-values obtained when applying the Wilcoxon, MC1-3 and the t-test respectively.


                            </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>


         <div class="containerNext">
            <div class="rowSpecialWithGap">
              <div class="colAll">

                  <h2 class="projectTitle">
                      Case example: immune cell association with Crohn’s disease
                  </h2>
                  <div class = 'example'>
                      <h4 class = 'exampleTitle'></h4>
                      <div class = 'exampleDesc'>

                      <p class="paragraph">
                      Moreover, we applied the method that performed the best (MC3) on a previously published experiment on the co-occurrence of Crohn's
                          disease-associated variants (GWAS SNPs) and DNase hypersensitive sites (DHSs) (published in Franke,
                          Andre, et al. "Genome-wide meta-analysis increases to 71 the number of confirmed Crohn's disease susceptibility loci." Nature genetics 42.12 (2010): 1118.).
                      </p>
                      <p class="paragraph">
                          The data processing and the filtering of the public available data can be found at <a href="../u/hbsuperuser/h/10---crohns-data-with-different-filtering"  target="_blank">
                          history</a>, while the results obtained by applying MC3 for the empirical analyses of the Crohn’s disease associated variants (GWAS SNPs) and DNase hypersensitive sites
                          (DHSs) in different cell types can be found at
                           <a href="../u/hbsuperuser/h/14-monte-carlo---mc3"  target="_blank">MC3 </a>.

                      </p>
                             </div>

                    </div>
                </div>
            </div>
         </div>

        <div class="footer">
            <div class = 'contactInfo'>
                <p class="paragraph">If you have any questions, requests or comments regarding the project, please contact: <a href="mailto:geirksa@ifi.uio.no">Geir Kjetil Sandve</a>.
            </div>
        </div>
    </div>

   <div id="menu1" class="tab-pane fade ">
        <div class="container">
            <div class="rowSpecial">
                <p class="colAllProjectList">
                    <h2 class="projectTitle">
                        General information about the Genomic HyperBrowser
                    </h2>
                    <p class="paragraph">
                        The main window (project page) is
                        divided into three parts; the first part (on the left) shows all the tools implemented in the
                        Genomic HyperBrowser. The tools related to this specific project are available under
                        the label 'BENCHMARKING PROJECT TOOLS'. The second part (middle) is the main resulting page.
                        The third part (on the right) shows all the
                        history elements of the actions made (run tools, uploaded data, etc).
                        On each of this history element one can run at least two operations:
                        <ol>
                            <li>by clicking on the icon 'eye' you can see the results of that element,
                            </li>
                            <li>by clicking on the title of the element you will see four new icons. Starting from the left
                                you have possibility of downloading the data, viewing the details, re-running the tool and visualizing
                                the results.
                            </li>
                            <li>However, if the user wants to re-run a specific element of the history, it is important that the history is imported first, by clicking
                                on the statement ‘Import history’ at the top right of the history page.
                            </li>
                        </ol>
                    </p>

                    <div class = 'example'>
                        <h4 class = 'exampleTitle'>Detailed example for re-running an history element:</h4>
                        <div class="main">
                            <p class="paragraph">
                                Let us assume that we want to re-run the analyses connected to the results for Benchmark 1 - Existence of an empirical correlation structure across case tracks and control tracks.
                            </p>
                        </div>


                            <p class="main">
                                <p class="paragraph">
                                    <div>
                                        <ol>
                                            <li>
                                                By clicking on the marked link, the history page called <b>‘[2.1]: GDT- Benchmark one’</b> will appear:
                                                <div class="">
                                                <img src="../static/project/step1.png"  style="width: 80%; margin-top: 15px;margin-bottom: 15px" />
                                            </div>
                                            </li>

                                            <li>
                                                Let’s now import the history, by clicking on ‘Import history (make a copy of this history and switch to it)'
                                                <div class="">
                                                    <img src="../static/project/step2.png"  style="width: 80%; margin-top: 10px;margin-bottom: 15px" />
                                                </div>
                                            </li>



                                            <li style="margin-bottom: 10px">The copy of the history will be generated and will appear (on the right part of the page):
                                                <div class="imgzoom">
                                                    <img src="../static/project/step3.png"  style="width: 80%; margin-top: 10px;margin-bottom: 15px" />
                                                </div>
                                            </li>

                                            <li>
                                                <div>
                                                    Let’s assume that we want to re-run the element <b>‘27: Benchmark One - T-test’</b>. For doing that, we need to click on the name of the element. Three new symbols will appear. By clicking on the third symbol starting from the right (‘run this job’):
                                                </div>

                                                <div class="imgzoom">
                                                    <img src="../static/project/step4.png"  style="width: 80%; margin-top: 10px;margin-bottom: 15px" />

                                                </div> The main page will change and look like the following:
                                                <div class="imgzoom">
                                                    <img src="../static/project/step5.png"  style="width: 80%; margin-top: 10px;margin-bottom: 15px" />
                                                </div>
                                                Now the element in the first selection box is the reference track, while in the second selection box, we have the case-control tracks.
                                                The third selection box shows the name of the column with case-controls labels.
                                                In this example we used as null model the 'T-test', the Forbes coefficient as similarity measure, the avarage as within group summary function and the right tailed test.

                                                <p>
                                                Now, we can ‘Execute’ the analysis by clicking on the ‘Execute’ command at the bottom of the page, and the new history element will start to run.
                                                </p>
                                                <p>
                                                The new history element is initially yellow while the job is running and will become green when the results are ready:
                                                </p>
                                                <div class="imgzoom">
                                                    <img src="../static/project/step6.png"  style="width: 80%; margin-top: 10px;margin-bottom: 15px" />
                                                </div>
                                            </li>

                                            <li style="margin-bottom: 10px">By clicking on the ‘eye’ of the element is possible to see the results of the analysis:
                                                <div class="imgzoom">
                                                    <img src="../static/project/step7.png"  style="width: 80%; margin-top: 10px;margin-bottom: 15px" />
                                                </div> The name of the element is <b>‘Group difference test - Benchmark One’</b>, however by clicking on the symbol ‘edit’ it is possible to rename the element depending on the analysis carried out, as we did for all the history elements.
                                            </li>

                                            <li>By clicking on ‘Import table to history (tabular)’ at the top of the table, a tabular format of the results is added to the history (element 31):
                                                <div class="imgzoom">
                                                    <img src="../static/project/step8.png"  style="width: 80%; margin-top: 10px;margin-bottom: 15px" />
                                                </div>
                                            </li>

                                            <li>Finally, the results can be download, by clicking on the name of the element and then on the download symbol:
                                                <div class="imgzoom">
                                                    <img src="../static/project/step9.png"  style="width: 80%; margin-top: 10px;margin-bottom: 15px" />
                                                </div>
                                                <div>
                                                    This last step, allows the user to download the data and plot the results by using the chosen software.
                                                </div>
                                            </li>
                                        </ol>
                                    </div>
                                </p>
                            </p>

                    </div>


            </div>
        </div>
    </div>

    <div id="menu2" class="tab-pane fade ">
        <div class="container">
            <div class="rowSpecial">
                <div class="colAllProjectList">

                      <div id ='content'></div>
                            <script>
                            $("#content").load("/dianadom_dev/static/hyperbrowser/files/projects/projects_sim-measure.txt");
                            </script>
                </div>
            </div>
        </div>
    </div>
    <div id="menu3" class="tab-pane fade ">
        <div class="container">


				<div class="contact">

				    <b>Note:</b>
                        <p>
                        This is a specialized version of The Genomic Hyperbrowser,
                        focusing on new functionality for analyzing collections of genomic track.
                        More information on the general HyperBrowser functionality can be
                        found on the <a href="https://hyperbrowser.uio.no/hb/" target="_blank">main project site</a>.
                        </p>
				    <b>Contact information</b>
                        <p>
                        If you have any questions, requests or comments, please use our discussion forum (available from the <a minsizehint="-1" href="https://sites.google.com/site/hyperbrowserhelp/?tool_id=hb_help">Help</a> menu) or send us an email to <a href="#" onclick="this.href='mailto:'+ this.innerHTML.split('').reverse().join('')" class="codedirection">hyperbrowser-requests@usit.uio.no</a>&#x200E;. We are happy to respond and welcome collaborative projects that help further develop our system. If you encounter any bugs, please let us know via an email to
                        <a href="#" onclick="this.href='mailto:'+ this.innerHTML.split('').reverse().join('')" class="codedirection">hyperbrowser-requests@usit.uio.no</a>.
                       </p>

				        <p>If you register as a user, you will be able to subscribe to the <a href="#" onclick="this.href='mailto:'+ this.innerHTML.split('').reverse().join('')" class="codedirection">hyperbrowser-requests@usit.uio.no</a> mailing list, which serves for distribution of system-related announcements and administrative messages.
				         A backlog of announcements is available from the Help menu. The Help menu also contains citation information.
				        </p>

				</div>

		</div>

        <div class="containerNext">
            <div class="rowSpecial">
                <div class ="infoAboutHB">
                <p class="blocktext">
                This project is being developed by the Norwegian bioinformatics community as an open-source project under the GPL  license v3,
                supported by various national and local bodies.
                </p>
                <p class="blocktext">
                  For further information on Norwegian bioinformatics, please visit <a href="http://www.bioinfo.no" target="_top">www.bioinfo.no</a>.
                </p>
         </div>
          </div>
		</div>

        <div class="footer">
                <div class="section">
                  <img src="hyperbrowser/images/logo/HB_support.png" class="center">
                </div>
        </div>

    </div>
    </div>
</div>



</body></html>