# Copyright (C) 2009, Geir Kjetil Sandve, Sveinung Gundersen and Morten Johansen
# This file is part of The Genomic HyperBrowser.
#
#    The Genomic HyperBrowser is free software: you can redistribute it and/or modify
#    it under the terms of the GNU General Public License as published by
#    the Free Software Foundation, either version 3 of the License, or
#    (at your option) any later version.
#
#    The Genomic HyperBrowser is distributed in the hope that it will be useful,
#    but WITHOUT ANY WARRANTY; without even the implied warranty of
#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#    GNU General Public License for more details.
#
#    You should have received a copy of the GNU General Public License
#    along with The Genomic HyperBrowser.  If not, see <http://www.gnu.org/licenses/>.

#This catalog is autogenerated by the WelcomePageBasic.py script
#Please DO NOT update it manually. The questions are defined in wpcontent.txt

from collections import OrderedDict

Catalog = OrderedDict([('1_1', '- Are the supplied trait-associated variants particularly active in certain cell types?'), ('1_2', '- Are the supplied trait-associated variants preferentially located in certain chromatin states (in a given cell type)?'), ('1_3', '- Are the supplied trait-associated variants potentially disrupting the regulatory function of a given transcription factor (in certain cell types)?'), ('1_4', '- How similar are the supplied traits in terms of sharing their lead SNPs?'), ('1_5', '- How do the supplied traits cluster based on the location of their lead SNPs?'), ('1_6', '- Which of the supplied traits is represented by the most (or the least) unique lead SNP set?'), ('1_7', '- What I am interested in does not match any of the suggestions above..'), ('2_1', '- Are the supplied somatic variants (e.g. variants associated with a particular cancer type) particularly active in certain cell types?'), ('2_2', '- Are the supplied somatic variants (e.g. variants associated with a particular cancer type) preferentially located in certain chromatin states (in a given cell type)?'), ('2_3', '- Are the supplied somatic variants (e.g. variants associated with a particular cancer type) potentially disrupting the regulatory function of certain transcription factors (in a given cell type)? (Example of a specialized tool.)'), ('2_4', '- Do the supplied somatic variants occur preferentially in certain genes?'), ('2_5', '- What I am interested in does not match any of the suggestions above..'), ('3_1', '- How do the supplied transcription factors cluster based on the location of their binding sites (i.e. based on the targets of their regulatory activity)?'), ('3_2', '- How much does binding (i.e. the set of supplied binding sites) of a given transcription factor vary across cell types?'), ('3_3', '- How similar are the supplied transcription factors in terms of sharing their binding sites?'), ('3_4', '- Which of the supplied transcription factors is represented by the most (or the least) unique binding site set?'), ('3_5', '- In which cell types do the supplied transcription factor binding sites fall into regulatory active regions?'), ('3_6', '- How do the supplied transcription factor binding sites overlap various traits\xe2\x80\x99 associated DNA variants?'), ('3_7', '- Which of the supplied genomic regions contain binding sites of a given transcription factor? (Example of a specialized tool.)'), ('3_8', '- Which transcription factors have binding sites located within the given genomic region(s)? (Example of a specialized tool.)')])

